src/test/java/org/apache/commons/io/input/ChecksumInputStreamTest.java

```diff
@@ -0,0 +1,163 @@
+            assertEquals(byteArray.length, checksum.getByteCount());
+            assertEquals(-4, checksum.getRemaining());
+        }
+    }
+
+    @Test
+    public void testReadTakingByteArrayThrowsException() throws IOException {
+        final Adler32 adler32 = new Adler32();
+        final byte[] byteArray = new byte[3];
+        final long sizeThreshold = -1859L;
+        try (ChecksumInputStream checksum = ChecksumInputStream.builder()
+        // @formatter:off
+                .setByteArray(byteArray)
+                .setChecksum(adler32)
+                .setExpectedChecksumValue((byte) -68)
+                .setCountThreshold(sizeThreshold)
+                .get()) {
+                // @formatter:on
+            assertEquals(0, checksum.getByteCount());
+            assertEquals(sizeThreshold, checksum.getRemaining());
+            // Ask to read one more byte than there is.
+            assertEquals(byteArray.length, checksum.read(new byte[byteArray.length + 1]));
+            // Next read is at EOF
+            assertThrows(IOException.class, () -> checksum.read(new byte[1]));
+            assertEquals(byteArray.length, checksum.getByteCount());
+            assertEquals(sizeThreshold - byteArray.length, checksum.getRemaining());
+        }
+    }
+
+    @Test
+    public void testReadTakingNoArgumentsThrowsException() throws IOException {
+        final CRC32 crc32 = new CRC32();
+        final byte[] byteArray = new byte[9];
+        try (ChecksumInputStream checksum = ChecksumInputStream.builder()
+        // @formatter:off
+                .setByteArray(byteArray)
+                .setChecksum(crc32)
+                .setExpectedChecksumValue((byte) 1)
+                .setCountThreshold(1)
+                .get()) {
+                // @formatter:on
+            assertEquals(0, checksum.getByteCount());
+            assertEquals(1, checksum.getRemaining());
+            assertThrows(IOException.class, () -> checksum.read());
+            assertEquals(1, checksum.getByteCount());
+            assertEquals(0, checksum.getRemaining());
+        }
+    }
+
+    @Test
+    public void testSkip() throws IOException {
+        // sanity-check
+        final CRC32 sanityCheck = new CRC32();
+        final byte[] byteArray = new byte[4];
+        sanityCheck.update(byteArray);
+        final long expectedChecksum = sanityCheck.getValue();
+        // actual
+        final CRC32 crc32 = new CRC32();
+        final InputStream byteArrayInputStream = new ByteArrayInputStream(byteArray);
+        try (ChecksumInputStream checksum = ChecksumInputStream.builder()
+        // @formatter:off
+                .setInputStream(byteArrayInputStream)
+                .setChecksum(crc32)
+                .setExpectedChecksumValue(expectedChecksum)
+                .setCountThreshold(33)
+                .get()) {
+                // @formatter:on
+            assertEquals(0, checksum.getByteCount());
+            @SuppressWarnings("unused")
```

Comment:
The warning doesn't need to be suppressed. Just remove the variable and leave the read.

Reply:
I replaced the suppression with an an assertion. Some static analysis  tools flag ignored read() values as a problem because it indicates a lack of EOF detection.